src/main/kotlin/no/nav/syfo/aksessering/api/SykmeldingApi.kt

```diff
@@ -36,13 +39,11 @@ fun Route.registerSykmeldingApi(sykmeldingService: SykmeldingService) {
             val sykmeldingsid = call.parameters["sykmeldingsid"]!!
             val principal: JWTPrincipal = call.authentication.principal()!!
             val subject = principal.payload.subject
-
             if (sykmeldingService.erEier(sykmeldingsid, subject)) {
-                if (sykmeldingService.registrerLestAvBruker(sykmeldingsid) > 0) {
-                    call.respond(HttpStatusCode.OK)
-                }
+                sykmeldingService.registrerStatus(SykmeldingStatusEvent(sykmeldingsid, LocalDateTime.now(), StatusEvent.CONFIRMED))
+                call.respond(HttpStatusCode.OK)
             } else {
-                call.respond(HttpStatusCode.Forbidden)
+                call.respond(HttpStatusCode.NotFound)
```

Comment:
Er det feil å oppgi at det er Forbidden, når den som gjør kallet ikkje eier sykmeldingen?

Reply:
Det er ikke feil, og for brukeren forsåker å få tilgang til en resource som den ikke har tilgang til så er det best å returnere en NotFound. Da den ikke finnes for brukeren og ved å gi en forbidden så sier man også at det finnes noe der